mw_url_rewrite.go

```diff
@@ -362,10 +364,13 @@ func checkHeaderTrigger(r *http.Request, options map[string]apidef.StringRegexMa
 		vals, ok := r.Header[mhCN]
 		if ok {
 			for i, v := range vals {
-				b := mr.Check(v)
-				if len(b) > 0 {
-					kn := "trigger-" + strconv.Itoa(triggernum) + "-" + mhCN + "-" + strconv.Itoa(i)
-					contextData[kn] = b
+				match := mr.FindStringSubmatch(v)
+				if len(match) > 0 {
+					kn := buildTriggerKey(triggernum, mhCN, i)
+					contextData[kn] = match[0]
+
+					addGroupsToContextData(&contextData, kn, match[1:])
```

Comment:
`match[1:]` looks a little suspicious - what if regex didn't have any groups? in worst case I would expect panic with index bound of range, in best case - call to function with no-op inside as `match[1:]` would give empty slice

Reply:
All expressions `match[1:]` placed inside of `if len(match) > 0 { .... }` blocks. 
If regexp doesn't have any groups it will result in `match` having only one item and `match[1:]` would be an empty slice with no-op call to function `addGroupsToContextData`
https://play.golang.org/p/-2KdJvgW8D9

@dencoded 
i'll wrap calls `addGroupsToContextData(&contextData, kn, match[1:])` into `if len(match) > 1 { .... }` - it doesn't affect functionality but increase readability